src/App.js

```diff
@@ -100,28 +103,32 @@ class App extends Component {
 // Helper function for requests
   requestCall(url, i) {
     const request = require('request');
-    const history = this.state.history.slice();
-    const current = history[history.length - 1];
-    const blocks = current.blocks.slice();
 
     request({
       uri: url,
-      method: 'GET'
+      method: 'GET',
     }, function(error, response, body) {
+      let result = null;
+
       if(error) {
-        blocks[i] = "OTHER";
+        result = "OTHER";
       } else if (response.statusCode === 200) {
-        blocks[i] = "UP";
+        result = "UP";
       } else {
         // alert("DOWN");
-        blocks[i] = "DOWN";
+        result = "DOWN";
       }
 
-      history[history.length - 1].blocks = blocks;
+      this.setState((prevState) => {
```

Comment:
So this is where it was going wrong. It's not an easy thing to fix and trips many people up.

Essentially there is a race condition between your dates to state.

For Example 
Test 1 finishes
Test 2 finishes

But when the function states they both save the current history to a variable and this history is the original starting history so each update just overwrites each other.